src/lib/compile-route.js

```diff
@@ -3,26 +3,32 @@ const express = require('path-to-regexp');
 const URL = require('url');
 const querystring = require('querystring');
 
-function normalizeRequest (url, options, Request) {
-	if (Request.prototype.isPrototypeOf(url)) {
+function normalizeRequest (input, options, Request) {
+	if (checkRequestInstance(input, Request)) {
+		const { url, method, headers } = input;
 		return {
-			url: url.url,
-			method: url.method,
+			url,
+			method,
 			headers: (() => {
-				const headers = {};
-				url.headers.forEach(name => headers[name] = url.headers.name);
-				return headers;
+				const mappedHeaders = {};
+				headers.forEach(name => mappedHeaders[name] = headers.name);
+				return mappedHeaders;
 			})()
 		};
 	} else {
 		return {
-			url: url,
+			url: input,
 			method: options && options.method || 'GET',
 			headers: options && options.headers
 		};
 	}
 }
 
+function checkRequestInstance (url, configRequest) {
```

Comment:
could you rename this function to `isRequestInstance` and also move it to above the function it's called from